zebra/dplane_fpm_nl.c

```diff
@@ -1031,9 +1031,39 @@ static int fpm_nl_finish(struct zebra_dplane_provider *prov, bool early)
 	struct fpm_nl_ctx *fnc;
 
 	fnc = dplane_provider_get_data(prov);
-	stream_free(fnc->ibuf);
-	stream_free(fnc->obuf);
-	close(fnc->socket);
+
+	/* This is the late call: stop the thread. */
+	if (early == false) {
```

Comment:
Hmm - maybe it's just a style nit, but I find the flow a little hard to follow. couldn't it be:
```
if (early) {
    do early stuff
    return 0;
} else {
    /* it's the real end */
    do final stuff
    return 0;
}
```

Reply:
IMO that looks the same to me... I think my initial code looks cleaner since it avoids an extra indentation level. If this bothers too much, maybe I'll split it into two functions.

Reply:
another nit, please don't compare bools to true / false, just test them directly; I did a mass conversion to fix all of these a while back and would like to keep it that way

Reply:
Fixed in last push.